frontend/src/metabase/visualizations/lib/settings/column.js

```diff
@@ -346,8 +346,8 @@ export const NUMBER_COLUMN_SETTINGS = {
     },
     default: true,
     getHidden: (column, settings, { series }) =>
-      settings["number_style"] !== "currency" ||
-      series[0].card.display !== "table",
+      settings["number_style"] !== "currency" &&
+      (!series[0].card.display || series[0].card.display === "table"),
```

Comment:
Rather than adjust this to say if display is undefined, It should be possible to get some information into that 3rd property in the getHidden function (sometimes called `extraData` in a few places). I think you could introduce something [here](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/admin/datamodel/metadata/components/FieldFormattingSettings/FieldFormattingSettings.tsx) into the `ColumnSettings` component that would make it's way there and help settings make a better decision